samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+
+                if (messages == null)
+                {
+                    log.LogInformation("No remaining messages found in queue.");
+                    break;
+                }
+
+                foreach (var message in messages)
+                {
+                    await receiver.CompleteAsync(message.SystemProperties.LockToken).ConfigureAwait(false);
+                    totalServiceBusMessages.Add(message);
+                }
+
+                if (messages.Count < MaxMessagesPerRequest)
+                {
+                    serviceBusMessagesRemaining = false;
+                }
+            }
+            while (serviceBusMessagesRemaining);
+
+            log.LogInformation($"Received {totalServiceBusMessages.Count} messages in total from queue.");
+
+            var validMessages = totalServiceBusMessages.Where(message => IsValidServiceBusMessage(message));
+
+            if (!validMessages.Any())
+            {
+                log.LogInformation("No valid messages were found in this function execution.");
+                return;
+            }
+
+            var validMessageCount = validMessages.Count();
+
+            log.LogInformation($"{validMessageCount} valid messages were found in this function execution.");
+            var chunkedMessages = new List<List<Message>>();
+
+            for (int i = 0; i < validMessageCount; i += MaxFilesPerTranscriptionJob)
+            {
+                var chunk = validMessages.Skip(i).Take(Math.Min(MaxFilesPerTranscriptionJob, validMessageCount - i)).ToList();
+                chunkedMessages.Add(chunk);
+            }
+
+            for (var i = 0; i < chunkedMessages.Count; i++)
+            {
+                await StartTranscriptionJob(chunkedMessages.ElementAt(i), log, i).ConfigureAwait(false);
+            }
+        }
+
+        internal static async Task<string> StartTranscriptionJob(IEnumerable<Message> messages, ILogger log, int index)
+        {
+            var locationString = string.Empty;
+            var subscriptionKey = CreateTranscriptionEnvironmentVariables.AzureSpeechServicesKey;
+            var subscriptionRegion = CreateTranscriptionEnvironmentVariables.AzureSpeechServicesRegion;
+            var locationUri = new Uri($"https://{subscriptionRegion}.api.cognitive.microsoft.com");
+            var subscription = new Subscription(subscriptionKey, locationUri);
+            var jobName = $"{StartDateTime.ToString("yyyy-MM-ddTHH:mm:ss", CultureInfo.InvariantCulture)}_{index}";
+            var locale = CreateTranscriptionEnvironmentVariables.Locale.Split('|')[0].Trim();
+
+            var serviceBusMessages = messages.Select(message => JsonConvert.DeserializeObject<ServiceBusMessage>(Encoding.UTF8.GetString(message.Body)));
+            var modelIds = new List<Guid>();
+
+            try
+            {
+                var properties = CreateTranscriptionPropertyBag();
+                modelIds = GetModelIds(log);
```

Comment:
GetModelIds from log? It looks weird. logs should just for tracing or debugging.